compiler/optimizer/LoopVersioner.cpp

```diff
@@ -8558,8 +8558,8 @@ bool TR_LoopVersioner::depsForLoopEntryPrep(
 
       traceMsg(comp(), "sverma: node: %p\n", node);
       // TODO_sverma: call TR::TransformUtil::findArrayBaseNode(comp(), node) for array base
-      TR::Node *base = arrayBaseAddressNode->getFirstChild();
-      if (base->isDataAddrPointer())
+      TR::Node *base = arrayBaseAddressNode;
+      if (base && base->isDataAddrPointer())
```

Comment:
Did we expand the paths that will get to this if block? Trying to understand why we didn't need to check base before.